aten/src/ATen/core/ivalue.h

```diff
@@ -960,6 +960,21 @@ inline bool IValue::isSameIdentity(IValue& rhs) {
         && this->payload.as_intrusive_ptr == rhs.payload.as_intrusive_ptr;
   }
 }
+
+inline bool shallowEquals(const IValue& lhs, const IValue& rhs) {
```

Comment:
why do we need shallow version? just to avoid comparing tensors?

Reply:
I'm not 100% confident how we want to define equality for IValues. We could make it shallow or we could make it deep and compare tensor contents, or anything inbetween. Note that deep comparison would be much harder for non-tensor Blobs with custom types.
So I took the easy way out and didn't try to define equality for IValues, but just defined a shallow equality, which is enough for my use case.